modules/tinymce/src/themes/silver/main/ts/ui/menus/menu/searchable/SearchableMenu.ts

```diff
@@ -0,0 +1,167 @@
+  // Consider moving some of these things into shared APIs. For example, make an extra API
+  // for TieredMenu to get the highlighted item.
+  return Sandboxing.getState(sandboxComp).bind(
+    Highlighting.getHighlighted
+  ).bind(
+    Highlighting.getHighlighted
+  );
+};
+
+const getSearchResults = (activeMenuComp: AlloyComponent): Optional<SugarElement<Element>> => {
+  // Depending on the menu layout, the search results will either be the entire
+  // menu, or something within the menu.
+  return Class.has(activeMenuComp.element, searchResultsClass)
+    ? Optional.some(activeMenuComp.element)
+    : SelectorFind.descendant(activeMenuComp.element, '.' + searchResultsClass);
+};
+
+// Model the interaction with ARIA
+const updateAriaOnHighlight = (tmenuComp: AlloyComponent, menuComp: AlloyComponent, itemComp: AlloyComponent): void => {
+  // This ARIA behaviour is based on the algolia example documented in TINY-8952
+  findWithinMenu(tmenuComp).each((inputComp) => {
+    setActiveDescendant(inputComp, itemComp);
+
+    const optActiveResults = getSearchResults(menuComp);
+    optActiveResults.each(
+      (resultsElem) => {
+        // Link aria-controls of the input to the id of the results container.
+        const controlledId = Attribute.get(resultsElem, 'id');
+        // Only set the ARIA attribute if we found a valid ID
+        if (controlledId !== undefined) {
+          Attribute.set(inputComp.element, 'aria-controls', controlledId);
+        }
+      }
+    );
+  });
+
+  // Update the aria-selected on the item. The removal is handled by onDehighlight
+  Attribute.set(itemComp.element, 'aria-selected', 'true');
+};
+
+const updateAriaOnDehighlight = (tmenuComp: AlloyComponent, menuComp: AlloyComponent, itemComp: AlloyComponent): void => {
+  // This ARIA behaviour is based on the algolia example documented in TINY-8952
+  Attribute.set(itemComp.element, 'aria-selected', 'false');
+};
+
+const focusSearchField = (tmenuComp: AlloyComponent): void => {
+  findWithinMenu(tmenuComp).each(
+    (searcherComp) => Focusing.focus(searcherComp)
+  );
+};
+
+const getSearchPattern = (dropdownComp: AlloyComponent): string => {
+  // Dropdowns are "coupled" with their sandbox and generally, create them on demand.
+  // When using "getExistingCoupled" of Coupling, it only returns the coupled
+  // component (here: the sandbox) if it already exists ... it won't do any creation.
+  // So here, we are trying to get possible fetchContext information for our fetch
+  // callback. If there is no sandbox, then there is no open menu, and we
+  // don't have any search context, so use an empty string. Otherwise, dive into
+  // the sandbox, and find the search field's current pattern.
+  const optSandboxComp = Coupling.getExistingCoupled(dropdownComp, 'sandbox');
+  return optSandboxComp.bind(
+    findWithinSandbox
+  ).map(saveState).map(
+    (state) => state.fetchPattern
+  ).getOr('');
```

Comment:
```suggestion
  return optSandboxComp
    .bind(findWithinSandbox)
    .map(saveState)
    .map((state) => state.fetchPattern)
    .getOr('');
```
As noted elsewhere this is normally the style we use for this as it's generally easier to read (it's like a list of steps).